### PR TITLE
Fixing #1561. Adding Intent->SerivceConnectionData mapping

### PR DESCRIPTION
**Pull request by menny**

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowApplicationTest.java [hunk 1]
```diff
@@ -138,14 +138,37 @@ public void packageManager_shouldKnowApplicationName() throws Exception {
   }
 
   @Test
+  public void bindServiceShouldCallOnServiceConnectedWithDefaultValues() {
+    TestService service = new TestService();
+    ComponentName expectedComponentName = new ComponentName("", "");
+    NullBinder expectedBinder = new NullBinder();
+    Shadows.shadowOf(RuntimeEnvironment.application).setComponentNameAndServiceForBindService(expectedComponentName, expectedBinder);
+    RuntimeEnvironment.application.bindService(new Intent(""), service, Context.BIND_AUTO_CREATE);
+    assertEquals(expectedComponentName, service.name);
```
Comment (coreydowning) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowApplicationTest.java]: Can you change the assertions to use `assertThat`? Thanks

**Comment by menny** [comment]
@coreydowning, done. I changed a few extra assertions too, just because they were there :wink:

**Comment by menny** [comment]
@coreydowning , didn't I do that with https://github.com/spotify/robolectric/commit/487e984e309394262147c6dbedd12895c84c09b2

**Comment by coreydowning** [comment]
Hmm I must be having github difficulties. Mind rebasing this and I'll get it merged?

**Comment by menny** [comment]
@coreydowning , done. I caused the merge conflict..

**Pull request merged**